src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs

```diff
@@ -768,6 +768,59 @@ public InstructionSetInfo(string name, string managedName, InstructionSet instru
             }
         }
 
+        public static IEnumerable<string> AllCpuFamilies()
+        {
+            // Only report these in the help, others are added to CpuFamilyToInstructionSets just for muscle memory after clang/llvm
+            yield return "generic";
+            yield return "nehalem";
+            yield return "sandybridge";
+            yield return "haswell";
+            yield return "ampere-altra";
+            yield return "apple-m1";
```

Comment:
Nit: Any chance we could make this a bit more metadata driven? I'm 99% sure we're going to forget updating this doc-only codepath when we add new macros.

If we shove it into some data structure that captures the name of the macro, the architecture it's allowed to expand on, whether it should show up in help, and the list of sets it expands into, we could have the structure drive both the help text and the runtime logic.